src/js/angular/resources/controllers.js

```diff
@@ -90,14 +94,17 @@ resourcesCtrl.controller('ResourcesCtrl', ['$scope', 'toastr', '$interval', '$ti
                         });
                     }
                 }
-            }).error(function (data) {
-                $scope.error = getError(data);
+            }).catch(function (error) {
+                $scope.error = getError(error);
                 $scope.loader = false;
+            }).finally(() => {
+                pendingRequest = false;
             });
         };
-
         const timer = $interval(function () {
```

Comment:
Few more thoughts here.
* Create a configuration or at least a constant for this 2000ms interval.
* What if given request takes say 3sec? Then you'll skip current iteration and in fact there will be a 4sec interval in the chart somewhere. Right? Is that OK? I guess that'd be an exceptional case probably but still.
* Why instead of having this `pendingRequest` flag you don't just start a simpler `setTimeout(getResourceMonitorData, 2000)` in the success handler.
* What's the requirement in case an error get's returned? In current implementation you'll just continue to poll the server and get errors. Is that OK? Can we consider some approach for handling this scenario? Say, if there are 5 requests without response and you stop polling and show some info in the widgets. Not sure if this wouldn't mean that   GDB is down for some reason.

Reply:
Extracted constant, removed `pendingRequest`